src/dev/flang/be/jvm/classfile/Expr.java

```diff
@@ -285,7 +586,7 @@ public static Expr commentAlways(String msg)
    * create invokestatic bytecode to call given class, name and descr producing
    * given result type on the stack.
    */
-  public static Expr invokeStatic(String cls, String name, String descr, JavaType rt)
+  public static Expr invokeStatic(String cls, String name, String descr, JavaType rt, int argCount)
```

Comment:
Instead of adding `argCount` as an argument here, try `ClassFileConstants.slotCountForArgs(descr)`.